### PR TITLE
Exclude dimensions used in faceting from squeeze

### LINKED ISSUE #7552: plotting facet grid with singleton dimension should create a facet grid with size 1
### What is your issue?

In my application, I want to plot facet grids of datasets, and the dimension I want to facet sometimes has a length of 1 and sometimes more than 1. When the length of the dimension is > 1, the `plot(col=....)` method of `DataArray` works as expected. However, when it is 1, I get an error and my code fails, although it would have been natrual for the plot method to create a `FacetGrid` with length of one facet.

As a minimal example, this code:
```python
import xarray as xr
da = xr.DataArray(np.random.rand(3,1), dims=('x', 'y'))
da.plot(col='y')
```
fails with an error:
```python
KeyError: 'y'
```
 This example works as expected:

```python
import xarray as xr
da = xr.DataArray(np.random.rand(3,2), dims=('x', 'y'))
da.plot(col='y')
```

### PR DESCRIPTION
**Pull request by wkrasnicki**
- [x] Closes #7552
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

Implements @headtr1ck solution from the issue.

### CODE REVIEW

**Review by headtr1ck — APPROVED** [review]
Thanks for the PR!

File: doc/whats-new.rst [hunk 1]
```diff
@@ -60,6 +60,9 @@ Bug fixes
   names were not updated properly internally (:issue:`7405`, :issue:`7588`,
   :pull:`8104`).
   By `Benoît Bovy <https://github.com/benbovy>`_.
+- Calling plot with kwargs ``col``, ``row`` or ``hue`` no longer results in an error.
```
Comment (mathause) [inline doc/whats-new.rst:63]:
```suggestion
- Calling plot with kwargs ``col``, ``row`` or ``hue`` no longer squeezes these dimensions.
```

Maybe a bit more specific?

File: doc/whats-new.rst [hunk 2]
```diff
@@ -60,6 +60,9 @@ Bug fixes
   names were not updated properly internally (:issue:`7405`, :issue:`7588`,
   :pull:`8104`).
   By `Benoît Bovy <https://github.com/benbovy>`_.
+- Calling plot with kwargs ``col``, ``row`` or ``hue`` no longer squeezes dimensions passed via these arguments.
```
Comment (mathause) [inline doc/whats-new.rst:63]:
```suggestion
- Calling plot with kwargs ``col``, ``row`` or ``hue`` no longer squeezes dimensions passed via these arguments
```

File: doc/whats-new.rst [hunk 3]
```diff
@@ -70,6 +70,11 @@ Bug fixes
   special case ``NaT`` handling in :py:meth:`~core.accessor_dt.DatetimeAccessor.isocalendar()`
   (:issue:`7928`, :pull:`8084`).
   By `Kai Mühlbauer <https://github.com/kmuehlbauer>`_.
+- Calling plot with kwargs ``col``, ``row`` or ``hue`` no longer results in an error.
+- Calling plot with kwargs ``col``, ``row`` or ``hue`` no longer squeezes dimensions passed via these arguments.
```
Comment (headtr1ck) [inline doc/whats-new.rst:74]: Only keep this line.
Some merge went wrong here

**Review by headtr1ck — APPROVED** [review]

File: doc/whats-new.rst (hunk 3 above)
Comment (dcherian) [inline doc/whats-new.rst:74]:
```suggestion
```

**Pull request merged**